addons/web_editor/controllers/main.py

```diff
@@ -270,130 +270,182 @@ def get_image_info(self, image_id=None, xml_id=None):
-                    bundle_2 = scss_files_data_by_bundle[j]
-                    # In unwanted bundles, keep only the files which are in wanted bundles too (_assets_helpers)
-                    if bundle_1[0]["xmlid"] not in bundles_restriction and bundle_2[0]["xmlid"] in bundles_restriction:
-                        bundle_1[1] = [item_1 for item_1 in bundle_1[1] if item_1 in bundle_2[1]]
-            for i in range(0, len(scss_files_data_by_bundle)):
-                bundle_1 = scss_files_data_by_bundle[i]
-                for j in range(i+1, len(scss_files_data_by_bundle)):
-                    bundle_2 = scss_files_data_by_bundle[j]
-                    # In every bundle, keep only the files which were not found in previous bundles
-                    bundle_2[1] = [item_2 for item_2 in bundle_2[1] if item_2 not in bundle_1[1]]
-
-            # Only keep bundles which still have files and that were requested
-            scss_files_data_by_bundle = [
-                data for data in scss_files_data_by_bundle
-                if (len(data[1]) > 0 and (not bundles_restriction or data[0]["xmlid"] in bundles_restriction))
-            ]
-
-            # Fetch the arch of each kept file, in each bundle
-            for bundle_data in scss_files_data_by_bundle:
-                for i in range(0, len(bundle_data[1])):
-                    url = bundle_data[1][i]
-                    url_info = url_infos[url]
-
-                    content = None
-                    if url_info["customized"]:
-                        # If the file is already customized, the content is found in the corresponding attachment
-                        content = base64.b64decode(custom_attachments.filtered(lambda a: a.url == url).datas)
-                    else:
-                        # If the file is not yet customized, the content is found by reading the local scss file
-                        module = url_info["module"]
-                        module_path = get_module_path(module)
-                        module_resource_path = get_resource_path(module, url_info["resource_path"])
-                        if module_path and module_resource_path:
-                            module_path = os.path.join(os.path.normpath(module_path), '') # join ensures the path ends with '/'
-                            module_resource_path = os.path.normpath(module_resource_path)
-                            if module_resource_path.startswith(module_path):
-                                with open(module_resource_path, "rb") as f:
-                                    content = f.read()
-
-                    bundle_data[1][i] = dict(
-                        url = "/%s/%s" % (url_info["module"], url_info["resource_path"]),
-                        arch = content,
-                        customized = url_info["customized"],
-                    )
+            scss_files_data_by_bundle = self.get_resources('scss', views, only_user_custom_files, bundles_restriction)
```

Comment:
Define `js_files_data_by_bundle` and `scss_files_data_by_bundle` outside their if, with an empty array (like it was before).